File: rok_3/aplikacje_mobilne/lista_1/Gabi_game_java/app/src/main/java/com/example/gabi_game_java/MainActivity.java
package com.example.gabi_game_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private int product;
    private long startTime;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setOperationAndButtons();
    }

    private void setOperationAndButtons() {
        Random rand = new Random(System.currentTimeMillis());
        int multiplicand = rand.nextInt(10);
        int multiplier = rand.nextInt(10);
        product = multiplicand * multiplier;

        TextView operation = (TextView) findViewById(R.id.operation);
        operation.setText(Integer.toString(multiplicand) + "\t\u22C5\t" + Integer.toString(multiplier));

        Button button1 = (Button) findViewById(R.id.button);
        button1.setOnClickListener(this::onClick);
        button1.setText(Integer.toString(rand.nextInt(100)));

        Button button2 = (Button) findViewById(R.id.button2);
        button2.setOnClickListener(this::onClick);
        button2.setText(Integer.toString(rand.nextInt(100)));

        Button button3 = (Button) findViewById(R.id.button3);
        button3.setOnClickListener(this::onClick);
        button3.setText(Integer.toString(rand.nextInt(100)));

        Button button4 = (Button) findViewById(R.id.button4);
        button4.setOnClickListener(this::onClick);
        button4.setText(Integer.toString(rand.nextInt(100)));

        switch (rand.nextInt(4)) {
            case 0:
                button1.setText(Integer.toString(product));
            case 1:
                button2.setText(Integer.toString(product));
            case 2:
                button3.setText(Integer.toString(product));
            case 3:
                button4.setText(Integer.toString(product));
        }
        startTime = System.currentTimeMillis();
    }

    private void onClick(View v) {
        long difference = System.currentTimeMillis() - startTime;

        Button button = (Button) v;
        if (Integer.parseInt(button.getText().toString()) == product) {
            Toast.makeText(this, Long.toString(difference) + " ms", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(this, "No, Mister", Toast.LENGTH_SHORT).show();
        }
        setOperationAndButtons();
    }
}

File: rok_3/aplikacje_mobilne/lista_1/Gabi_game_kotlin/app/src/main/java/com/example/gabi_game_kotlin/MainActivity.kt
package com.example.gabi_game_kotlin

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.gabi_game_kotlin.databinding.ActivityMainBinding
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var product = 0
    private var startTime: Long = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        setOperationAndButtons()
    }

    fun setOperationAndButtons() {
        val rand = Random
        val multiplicand = rand.nextInt(10)
        val multiplier = rand.nextInt(10)
        product = multiplicand * multiplier

        binding.operation.text = multiplicand.toString() + "\t\u22C5\t" + multiplier.toString()
        binding.button.text = rand.nextInt(100).toString()
        binding.button2.text = rand.nextInt(100).toString()
        binding.button3.text = rand.nextInt(100).toString()
        binding.button4.text = rand.nextInt(100).toString()

        binding.button.setOnClickListener(View.OnClickListener { v: View? -> onClick(v!!) })
        binding.button2.setOnClickListener(View.OnClickListener { v: View? -> onClick(v!!) })
        binding.button3.setOnClickListener(View.OnClickListener { v: View? -> onClick(v!!) })
        binding.button4.setOnClickListener(View.OnClickListener { v: View? -> onClick(v!!) })

        when (rand.nextInt(4)) {
            0 -> {
                binding.button.text = product.toString()
            }
            1 -> {
                binding.button2.text = product.toString()

            }
            2 -> {
                binding.button3.text = product.toString()

            }
            3 -> {
                binding.button4.text = product.toString()
            }
        }

        startTime = System.currentTimeMillis()
    }

    fun onClick(view: View) {
        val difference = System.currentTimeMillis() - startTime

        val button = view as Button
        if (button.text.toString().toInt() == product) {
            Toast.makeText(this, difference.toString() + " ms", Toast.LENGTH_SHORT).show()
        } else {
            Toast.makeText(this, "No, Mister", Toast.LENGTH_SHORT).show()
        }
        setOperationAndButtons()
    }
}